test/eredis_cluster_tests.erl

```diff
@@ -384,6 +386,48 @@ basic_test_() ->
     }
 }.
 
+test_command_support() ->
+    %% Test that we can extract the key from various commands. We only care
+    %% about the first key in a command.
+    Commands =
+        [
+         %% Default: Key is 2nd (directly after the command name)
+         ["xadd", "k", "MAXLEN", "42", "*", "foo", "bar"],
+         ["set", "k", "value"],
+
+         %% Key is 3rd (after subcommand)
+         ["xinfo", "stream", "k"],
+         ["xinfo", "consumers", "k", "somegroup"],
+         ["xgroup", "create", "k", "somegroup", "$"],
+         ["XGROUP", "CREATECONSUMER", "k", "somegroup", "myconsumer123"],
+         ["memory", "usage", "k"],
+         ["object", "encoding", "k"],
+         %% New in Redis 6.2 (not yet used in builds):
+         %%["zunion", "2", "k", "k2"], % Added in Redis 6.2
+         %%["zinter", "2", "k", "k2"], % Added in Redis 6.2
+         %%["zdiff", "2", "k", "k2"], % Added in Redis 6.2
+
+         %% Key is 4th
+         ["eval", "return redis.call('set',KEYS[1],'bar')", "1", "k"],
+         ["evalsha", "adc98b7d8c89adc87b8dca98", "1", "k"],
+
+         %% Key is after the STREAMS keyword
+         ["xread", "count", "42", "streams", "k", "k2", "1-0", "1-0"],
+         ["xreadgroup", "GROUP", "gr", "my-consumer", "COUNT", "42",
+          "STREAMS", "k", "foo", "42-0", "42-0"]
+        ],
```

Comment:
Do we add any test of the multi command ?
Possibly covered by a transaction call in a test?

Reply:
Supposedly covered by transaction test cases as you say. I didn't test any pipelines (lists of commands) now.